tools/bin/identify-dormant-workflows.py

```diff
@@ -0,0 +1,73 @@
+import argparse
+import os
+import subprocess
+import re
+import json
+from datetime import datetime, timedelta
+from github import Github
+from slack_sdk import WebClient
+from slack_sdk.errors import SlackApiError
+
+DAYS_TO_KEEP_ORPHANED_JOBS = 90
+SLACK_CHANNEL_FOR_NOTIFICATIONS = "infra-alerts"
+
+# Initiate the parser
+parser = argparse.ArgumentParser()
+
+# Add long and short argument
+parser.add_argument("--pat", "-p", help="Set github personal access token")
+parser.add_argument("--sat", "-s", help="Set slack api token")
+
+def main():
+    # Read arguments from the command line
+    args = parser.parse_args()
+    # Check for user supplied PAT. If not supplied, assume we are running in actions
+    # and pull from environment
+
+    gh_token = None
+    slack_token = None
+
+    if args.pat:
+        gh_token = args.pat
+    else:
+        gh_token = os.getenv('GITHUB_TOKEN')
+    if not gh_token:
+        raise Exception("Github personal access token not provided via args and not available in GITHUB_TOKEN variable")
+    
+    if args.sat:
+        slack_token = args.sat
+    else:
+        slack_token = os.getenv('SLACK_TOKEN')
+    if not slack_token:
+        raise Exception("Slack app token not provided via args and not available in SLACK_TOKEN variable")
```

Comment:
Is there a use case where we don't care about slack for this?  I feel like on one hand it would be nice to run locally, but I think running in prod it seems like we should get pretty easy signal anyway.  ¯\\_(ツ)_/¯

Reply:
I'll make this optional and it will just not send the notification if not passed.